packages/checkout-storefront/src/hooks/useCheckoutFormValidationTrigger.ts

```diff
@@ -1,38 +1,43 @@
+import { FormDataBase, hasErrors, UseFormReturn } from "@/checkout-storefront/hooks/useForm";
 import {
   CheckoutFormScope,
   useCheckoutValidationActions,
   useCheckoutValidationState,
 } from "@/checkout-storefront/state/checkoutValidationStateStore";
-import { FormDataBase } from "@/checkout-storefront/lib/globalTypes";
 import { useCallback, useEffect } from "react";
-import { UseFormReturn } from "react-hook-form";
 
 interface UseCheckoutFormValidationTriggerProps<TData extends FormDataBase> {
   scope: CheckoutFormScope;
-  formProps: Pick<UseFormReturn<TData>, "trigger" | "formState">;
+  form: UseFormReturn<TData>;
 }
 
 // tells forms to validate once the pay button is clicked
 export const useCheckoutFormValidationTrigger = <TData extends FormDataBase>({
   scope,
-  formProps,
+  form,
 }: UseCheckoutFormValidationTriggerProps<TData>) => {
   const { setValidationState } = useCheckoutValidationActions();
   const { validating } = useCheckoutValidationState();
 
-  const { trigger } = formProps;
+  const { values, validateForm, setTouched } = form;
 
   const handleGlobalValidationTrigger = useCallback(async () => {
     if (validating) {
-      const formValid = await trigger();
-      if (formValid) {
+      const formErrors = await validateForm(values);
+      if (!hasErrors(formErrors)) {
         setValidationState(scope, "valid");
         return;
       }
 
+      const touched = Object.keys(formErrors).reduce(
+        (result, key) => ({ ...result, [key]: true }),
+        {}
+      );
+
+      void setTouched(touched, true);
```

Comment:
Not really, in e.g text input we manually do `touched[field.name] ? (errors[field.name] as string) : undefined` because formik treats those two props as separate stuff. Plus tested for this scenario (so the user touches nothing, just clicks pay and all the fields turn red) and it works fine